src/main/kotlin/com/openlattice/chronicle/controllers/TimeUseDiaryController.kt

```diff
@@ -73,30 +78,35 @@ class TimeUseDiaryController(
         @PathVariable(PARTICIPANT_ID) participantId: String,
         @RequestBody responses: List<TimeUseDiaryResponse>
     ): UUID {
-        ensureAuthenticated()
-        val hds = storageResolver.getPlatformStorage(PostgresFlavor.VANILLA)
-        val timeUseDiaryId = idGenerationService.getNextId()
-        AuditedOperationBuilder<Unit>(hds.connection, auditingManager)
-            .operation { connection ->
-                timeUseDiaryManager.submitTimeUseDiary(
-                    connection,
-                    timeUseDiaryId,
-                    organizationId,
-                    studyId,
-                    participantId,
-                    responses
-                ) }
-            .audit { listOf(
-                AuditableEvent(
-                    AclKey(timeUseDiaryId),
-                    Principals.getCurrentSecurablePrincipal().id,
-                    Principals.getCurrentUser(),
-                    AuditEventType.SUBMIT_TIME_USE_DIARY,
-                    ""
-                )
-            ) }
-            .buildAndRun()
-        return timeUseDiaryId
+
+        var maybeRealStudyId: UUID? = null
+        if (studyService.isLegacyStudyId(studyId)) {
+            maybeRealStudyId = studyService.getRealStudyIdForLegacyStudyId(studyId)
+        }
+        val realStudyId = maybeRealStudyId ?: studyId
+
+        return storageResolver.getPlatformStorage().connection.use { conn ->
+            AuditedOperationBuilder<UUID>(conn, auditingManager)
+                .operation { connection ->
+                    timeUseDiaryService.submitTimeUseDiary(
+                        connection,
+                        organizationId,
+                        realStudyId,
+                        participantId,
+                        responses
+                    )
+                }
+                .audit {
+                    listOf(
+                        AuditableEvent(
+                            aclKey = AclKey(realStudyId),
+                            study = realStudyId,
+                            eventType = AuditEventType.SUBMIT_TIME_USE_DIARY,
```

Comment:
if not provided, the defaults for AuditableEvent uses principals and ids from the currentUser

Reply:
Yea ... waiting on your changes for that!